File: problems/2210-count-hills-and-valleys-in-an-array/solution.cpp
class Solution {
public:
    int countHillValley(vector<int>& nums) {
        int count = 0;
        for (int i = 1; i < nums.size() - 1; ++i) {
            int left = i - 1;
            while (left >= 0 && nums[left] == nums[i]) {
                left--;
            }
            int right = i + 1;
            while (right < nums.size() && nums[right] == nums[i]) {
                right++;
            }
            if (left >= 0 && right < nums.size()) {
                if ((nums[i] > nums[left] && nums[i] > nums[right]) || (nums[i] < nums[left] && nums[i] < nums[right])) {
                    count++;
                }
            }
        }
        return count;
    }
};

File: problems/2210-count-hills-and-valleys-in-an-array/solution.py
class Solution:
    def countHillValley(self, nums: List[int]) -> int:
        count = 0
        for i in range(1, len(nums) - 1):
            left = i - 1
            right = i + 1
            while left >= 0 and nums[left] == nums[i]:
                left -= 1
            while right < len(nums) and nums[right] == nums[i]:
                right += 1
            if left >= 0 and right < len(nums):
                if (nums[i] > nums[left] and nums[i] > nums[right]) or \
                   (nums[i] < nums[left] and nums[i] < nums[right]):
                    count += 1
        return count